### PR TITLE
Build: Update scripts for release (fixes #475)

### LINKED ISSUE #475: espree v8 release process
After we merge https://github.com/eslint/espree/pull/469, we should update the release process to account for an additional build step.

The new build step (`npm run rollup` in https://github.com/eslint/espree/pull/469) does two things:

1. Reads the _new_ version from `package.json` to update `lib/version.js`.
2. Creates `dist/espree.cjs`.

Current steps in [generateRelease](https://github.com/eslint/eslint-release/blob/c1391a40b633315ee7b79cffad7043491f9e44f9/lib/release-ops.js#L344) are:

1. `npm test`.
2. Update the changelog and commit that change.
3. `npm version` to update `package.json`. commit that change, and create git tag.

But now we should have something like this:

1. Update the changelog and commit that change.
2. `npm version` to update `package.json`. commit that change, and create git tag.
3. `npm run rollup`.
4. `npm test`

An additional question is whether or not should we commit `lib/version.js`. It might make sense to have a fixed `"master"` (or `"main"`) as the version string in the repo.

Previous discussion: https://github.com/eslint/espree/pull/469#discussion_r584326147

### PR DESCRIPTION
**Pull request by nzakas**
1. Split updating of `lib/version.js` into a separate script and created `update-version` npm script
1. Renamed `rollup` script to `build` to be a bit more conventional
2. Added a `pretest` script to run `build` before `npm test`
3. Added a `prepublishOnly` script to run `update-version` and then `build` before `npm publish`

I believe this should take care of any of our remaining issues with the release process. We should, of course, run an alpha release first to verify. I tested this locally and everything seemed to work.

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]

**Review by btmills — APPROVED** [review]
I tested this locally as well, and it worked correctly as far as I could tell.

File: package.json [hunk 1]
```diff
@@ -74,7 +74,10 @@
     "test": "npm-run-all -p unit lint",
     "lint": "eslint \"*.?(c)js\" lib/ tests/lib/",
     "fixlint": "npm run lint -- --fix",
-    "rollup": "rollup -c rollup.config.js",
+    "build": "rollup -c rollup.config.js",
```
Comment (mdjermanovic) [inline package.json:77]: We should update [ci.yml](https://github.com/eslint/espree/blob/main/.github/workflows/ci.yml), it uses `npm run rollup`.

**Review by mdjermanovic — COMMENTED** [review]
Looks good to me,

**Comment by nzakas** [comment]
@mdjermanovic nice catch. Updated.

**Review by mdjermanovic — APPROVED** [review]
LGTM!

**Pull request merged**